app/controllers/support_interface/validation_errors_controller.rb

```diff
@@ -1,18 +1,38 @@
 module SupportInterface
   class ValidationErrorsController < SupportInterfaceController
+    include ValidationErrorHelper
     def index
+      @filter_form = FilterValidationForm.new
       @grouped_counts = ValidationError.group(:form_object).order("count_all DESC").count
+      @form_objects_select = @grouped_counts.map {|key, _value|
+ OpenStruct.new(id: key, name: humanized_form_object(key)) }.append(OpenStruct.new(id: "", name: "All"))
+
       @grouped_column_error_counts = ValidationError.list_of_distinct_errors_with_count
+      @attribute_select = @grouped_column_error_counts.map {|column|
+ OpenStruct.new(id: column[0][1], 
+name: column[0][1].underscore.titleize.gsub('.', ' ')) }.append(OpenStruct.new(id: "", name: "All"))
+    end
+
+    def filter
+      redirect_to history_support_interface_validation_errors_path(form_object: filter_params[:form_object], 
+attribute: filter_params[:attribute])
     end
 
     def history
       @form_object = params[:form_object]
       @attribute = params[:attribute]
-
-      @errors = ValidationError.where(form_object: @form_object).order(created_at: :desc)
+      @errors = if !@form_object.empty?
```

Comment:
are we able to not use a double negative?
```suggestion
      @errors = if @form_object.present?
```